datasources/beeper/beeper.go

```diff
@@ -0,0 +1,982 @@
+/*
+	Timelinize
+	Copyright (c) 2024 Timelinize Contributors
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published
+	by the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+// Package beeper implements a data source for Beeper chat data.
+package beeper
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/timelinize/timelinize/timeline"
+	"go.uber.org/zap"
+)
+
+// Data source name and ID.
+const (
+	DataSourceName = "Beeper"
+	DataSourceID   = "beeper"
+)
+
+func init() {
+	err := timeline.RegisterDataSource(timeline.DataSource{
+		Name:            DataSourceID,
+		Title:           DataSourceName,
+		Icon:            "beeper.svg",
+		Description:     "Beeper chat messages importer",
+		NewFileImporter: func() timeline.FileImporter { return new(Beeper) },
+	})
+	if err != nil {
+		timeline.Log.Fatal("registering data source", zap.Error(err))
+	}
+}
+
+// Beeper interacts with the Beeper SQLite database to extract chat messages.
+type Beeper struct{}
+
+// Recognize returns whether the input file is recognized as a Beeper database.
+func (Beeper) Recognize(_ context.Context, dirEntry timeline.DirEntry, _ timeline.RecognizeParams) (timeline.Recognition, error) {
+	filename := filepath.Base(dirEntry.Name())
+	// Check for Beeper database filenames
+	if filename == "index.db" || filename == "contacts.db" {
+		var ok bool
+		var err error
+		if ok, err = checkTables(dirEntry.FullPath()); err != nil {
+			return timeline.Recognition{}, fmt.Errorf("checking table existence: %w", err)
+		}
+
+		if ok {
+			return timeline.Recognition{Confidence: 1}, nil
+		}
+	}
+
+	return timeline.Recognition{}, nil
+}
+
+// FileImport conducts an import of the Beeper data.
+func (b *Beeper) FileImport(ctx context.Context, dirEntry timeline.DirEntry, params timeline.ImportParams) error {
+	path := dirEntry.FullPath()
+	params.Log.Info("Starting Beeper import", zap.String("file", path))
+
+	// Get user's home directory
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("getting home directory: %w", err)
+	}
+
+	// Try standard Beeper location first (most reliable)
```

Comment:
`FileImport()` should import the named dir entry; finding the data on the system isn't its job. That should be left to the import planner + `Recognize()`.